homeassistant/components/tensorflow/image_processing.py

```diff
@@ -58,49 +66,72 @@
         vol.Optional(CONF_FILE_OUT, default=[]): vol.All(cv.ensure_list, [cv.template]),
         vol.Required(CONF_MODEL): vol.Schema(
             {
-                vol.Required(CONF_GRAPH): cv.isfile,
+                vol.Required(CONF_GRAPH): cv.isdir,
                 vol.Optional(CONF_AREA): AREA_SCHEMA,
                 vol.Optional(CONF_CATEGORIES, default=[]): vol.All(
                     cv.ensure_list, [vol.Any(cv.string, CATEGORY_SCHEMA)]
                 ),
                 vol.Optional(CONF_LABELS): cv.isfile,
+                vol.Optional(CONF_LABEL_OFFSET, default=1): int,
                 vol.Optional(CONF_MODEL_DIR): cv.isdir,
             }
         ),
     }
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+def get_model_detection_function(model):
+    """Get a tf.function for detection."""
+
+    @tf.function
+    def detect_fn(image):
+        """Detect objects in image."""
+
+        image, shapes = model.preprocess(image)
+        prediction_dict = model.predict(image, shapes)
+        detections = model.postprocess(prediction_dict, shapes)
+
+        return detections
+
+    return detect_fn
+
+
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the TensorFlow image processing platform."""
     model_config = config.get(CONF_MODEL)
     model_dir = model_config.get(CONF_MODEL_DIR) or hass.config.path("tensorflow")
     labels = model_config.get(CONF_LABELS) or hass.config.path(
         "tensorflow", "object_detection", "data", "mscoco_label_map.pbtxt"
     )
+    checkpoint = os.path.join(model_config.get(CONF_GRAPH), "checkpoint")
+    pipeline_config = os.path.join(model_config.get(CONF_GRAPH), "pipeline.config")
 
     # Make sure locations exist
-    if not os.path.isdir(model_dir) or not os.path.exists(labels):
-        _LOGGER.error("Unable to locate tensorflow models or label map")
+    if (
+        not os.path.isdir(model_dir)
```

Comment:
This does I/O and must be run within the executor.

Is there a reason why we're moving to async context and using `async_setup_platform` instead of `setup_platform`?